examples/music-bot/src/bot.ts

```diff
@@ -0,0 +1,173 @@
+client.on('message', async (message) => {
+	if (!message.guild) return;
+	if (!client.application?.owner) await client.application?.fetch();
+
+	if (message.content.toLowerCase() === '!deploy' && message.author.id === client.application?.owner?.id) {
+		await message.guild.commands.create({
+			name: 'play',
+			description: 'Plays a song',
+			options: [
+				{
+					name: 'song',
+					type: 'STRING' as const,
+					description: 'The URL of the song to play',
+					required: true,
+				},
+			],
+		});
+
+		await message.guild.commands.create({
+			name: 'skip',
+			description: 'Skip to the next song in the queue',
+		});
+
+		await message.guild.commands.create({
+			name: 'queue',
+			description: 'See the music queue',
+		});
+
+		await message.guild.commands.create({
+			name: 'pause',
+			description: 'Pauses the song that is currently playing',
+		});
+
+		await message.guild.commands.create({
+			name: 'resume',
+			description: 'Resume playback of the current song',
+		});
+
+		await message.guild.commands.create({
+			name: 'leave',
+			description: 'Leave the voice channel',
+		});
+
+		await message.reply('Deployed!');
+	}
+});
+
+/**
+ * Maps guild IDs to music subscriptions, which exist if the bot has an active VoiceConnection to the guild.
+ */
+const subscriptions = new Map<Snowflake, MusicSubscription>();
+
+// Handles slash command interactions
+client.on('interaction', async (interaction: Interaction) => {
+	if (!interaction.isCommand() || !interaction.guildID) return;
+	let subscription = subscriptions.get(interaction.guildID);
+
+	if (interaction.commandName === 'play') {
+		// Extract the video URL from the command
+		const url = interaction.options.get('song')!.value! as string;
+
+		// If a connection to the guild doesn't already exist and the user is in a voice channel, join that channel
+		// and create a subscription.
+		if (!subscription) {
+			if (interaction.member instanceof GuildMember && interaction.member.voice.channel) {
+				const channel = interaction.member.voice.channel;
+				subscription = new MusicSubscription(
+					joinVoiceChannel({
+						channelId: channel.id,
+						guildId: channel.guild.id,
+						adapterCreator: createDiscordJSAdapter(channel),
+					}),
+				);
+				subscription.voiceConnection.on('error', console.warn);
+				subscriptions.set(interaction.guildID, subscription);
+			}
+		}
+
+		// If there is no subscription, tell the user they need to join a channel.
+		if (!subscription) {
+			await interaction.reply('Join a voice channel and then try that again!');
+			return;
+		}
+
+		// Make sure the connection is ready before processing the user's request
+		try {
+			await entersState(subscription.voiceConnection, VoiceConnectionStatus.Ready, 20e3);
+		} catch (error) {
+			console.warn(error);
+			await interaction.reply('Failed to join voice channel within 20 seconds, please try again later!');
```

Comment:
Since the interaction token times out after 3 seconds if you don't defer, you should probably defer first.